docc.go

```diff
@@ -2,168 +2,226 @@ package docc
+			strings.Contains(file.Name, "footer") ||
+			strings.Contains(file.Name, "footnotes") {
+			openedFile, err := zipReadCloser.Open(file.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			fileReaders = append(fileReaders, FileReader{
+				fileName: file.Name,
+				xml:      openedFile,
+				decoder:  xml.NewDecoder(openedFile),
+			})
+		}
+	}
+
+	reader.fileReaders = fileReaders
+
+	return reader, nil
 }
 
 // NewReader generetes a Reader struct.
 // After reading, the Reader struct shall be Close().
 func NewReader(docxPath string) (*Reader, error) {
-	r := new(Reader)
-	r.docxPath = docxPath
+	reader := new(Reader)
+	reader.docxPath = docxPath
 	ext := strings.ToLower(filepath.Ext(docxPath))
-	if ext == ".doc" {
-		xfp, err := docToX(docxPath)
-		if err != nil {
-			return nil, fmt.Errorf("fail or not support .doc file: %w", err)
-		}
-		r.docxPath = xfp
-		r.fromDoc = true
-	} else if ext != ".docx" {
+	if ext != ".docx" {
 		return nil, ErrNotSupportFormat
 	}
 
-	a, err := zip.OpenReader(r.docxPath)
+	zipReadCloser, err := zip.OpenReader(reader.docxPath)
 	if err != nil {
 		return nil, err
 	}
-	r.docx = a
 
-	f, err := a.Open("word/document.xml")
-	if err != nil {
-		return nil, err
+	var fileReaders []FileReader
+	for _, file := range zipReadCloser.File {
+		if file.Name == "word/document.xml" ||
+			strings.Contains(file.Name, "header") ||
+			strings.Contains(file.Name, "footer") ||
+			strings.Contains(file.Name, "footnotes") {
+			openedFile, err := zipReadCloser.Open(file.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			fileReaders = append(fileReaders, FileReader{
+				fileName: file.Name,
+				xml:      openedFile,
+				decoder:  xml.NewDecoder(openedFile),
+			})
+		}
 	}
-	r.xml = f
-	r.dec = xml.NewDecoder(f)
 
-	return r, nil
+	reader.fileReaders = fileReaders
+	reader.docx = zipReadCloser
+
+	return reader, nil
 }
 
-// Read reads the .docx file by a paragraph.
-// When no paragraphs are remained to read, io.EOF error is returned.
-func (r *Reader) Read() (string, error) {
-	err := seekNextTag(r.dec, "p")
+func (r *Reader) read(decoder *xml.Decoder) (string, error) {
+	err := seekNextTag(decoder, xmlParagraph)
 	if err != nil {
 		return "", err
 	}
-	p, err := seekParagraph(r.dec)
+	paragraph, err := seekParagraph(decoder)
 	if err != nil {
 		return "", err
 	}
-	return p, nil
+	return paragraph, nil
 }
 
-// ReadAll reads the whole .docx file.
-func (r *Reader) ReadAll() ([]string, error) {
-	ps := []string{}
+func (r *Reader) readSingleFile(decoder *xml.Decoder) (string, error) {
+	var content strings.Builder
 	for {
-		p, err := r.Read()
+		paragraph, err := r.read(decoder)
 		if err == io.EOF {
-			return ps, nil
+			return content.String(), nil
 		} else if err != nil {
-			return nil, err
+			return "", err
+		}
+		content.WriteString(paragraph)
+		content.WriteString(" ")
+	}
+}
+
+func (r *Reader) ReadAllFiles() (headerValue, contentValue, footerValue string, err error) {
```

Comment:
Might be useful to have documentation that explains what is being read, and how the data is formatted.